src/app/components/UserEventDetails.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import type { IEvent } from "../../database/eventSchema";
+import style from "@styles/UserEventDetails.module.css";
+
+type IParams = {
+  id: string;
+};
+
+export default function UserEventDetails({ id }: IParams) {
+  const [eventData, setEventData] = useState<IEvent | null>(null);
+
+  async function fetchEventData() {
+    try {
+      const response = await fetch(`http://localhost:3000/api/event/${id}`);
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch event. Status: ${response.status}`);
+      }
+
+      const data = await response.json();
+      setEventData(data);
+    } catch (err: unknown) {
+      console.error("Error:", err);
+      setEventData(null);
+    }
+  }
+
+  useEffect(() => {
+    fetchEventData();
+  }, [id]);
+
+  return (
+    <>
+      {eventData ? (
+        <div className={style.modal}>
+          <h1 className={style.eventName}>{eventData.name}</h1>
+          <div className={style.details}>
+            <h2 className={style.info}>
+              Date: {new Date(eventData.date).toDateString()}
```

Comment:
I believe the date field should always be a Date according to our Event schema. However I think there is some finicky stuff happening when Mongo takes in dates and sends them back out, which is why we have to cast it back to a Date object with new Date().